### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by hridambasu**
Updated readme file by adding instructions on how to build and run. Also added pointers for benchmarks and examples.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,15 +13,20 @@ This codebase is generally intended to be network-agnostic. Programs take messag
 ## Project Dependencies
 This project relies on the `libpaillier` Rust crate using the GMP backend. GMP should be available during build-time.  
 
-##  What's Implemented
+## Rust Dependencies and Versions
+
+The preferred Rust version is cargo 1.68.2.
```
Comment (amaloz) [inline README.md:18]: If we have a preferred rust version, should we instead pin it using the `rust-toolchain` file?

File: README.md [hunk 2]
```diff
@@ -13,15 +13,20 @@ This codebase is generally intended to be network-agnostic. Programs take messag
 ## Project Dependencies
 This project relies on the `libpaillier` Rust crate using the GMP backend. GMP should be available during build-time.  
 
-##  What's Implemented
+## Rust Dependencies and Versions
+
+The preferred Rust version is cargo 1.68.2. 
+The preferred GMP version is 6.2.1.
+
+###  What's Implemented
```
Comment (amaloz) [inline README.md:21]: I think this has one extra heading level, should be ## instead?

File: README.md [hunk 3]
```diff
@@ -13,15 +13,20 @@ This codebase is generally intended to be network-agnostic. Programs take messag
...
 
 ### Key Generation (Figure 5 of CGGMP20)
 
 KeyGen generates a threshold signing key, shares of which are distributed to each node. Every node outputs a private key along with the public keys of all other nodes. This only needs to be run once for a given set of nodes. 
 
 ### Auxinfo (CGGMP20 Figure 6, minus the key refreshing)
 
-Auxinfo generates the auxilary information (Paillier keys and ring-Pedersen parameters) needed in order to compute presignatures. In CGGMP20, this is done in parallel with key refreshing, however this codebase currently only implements the generation of auxilary information. This is run after KeyGen and only needs to be run once.
+Auxinfo generates the auxilary information (Paillier keys and ring-Pedersen parameters) needed in order to compute presignatures. In CGGMP20, this is done in parallel with key refreshing, however this codebase currently only implements the generation of auxillary information. This is run after KeyGen and only needs to be run once.
```
Comment (amaloz) [inline README.md:29]: I think the correct spelling is "auxiliary" :-)

**Review by amaloz — COMMENTED** [review]
Some minor comments / suggestions.

File: README.md [hunk 4]
```diff
@@ -39,6 +44,38 @@ Currently, the codebase only implements n-out-of-n sharing. While t-out-of-n sha
...
+
+While some thought has been put into handling invalid messages (duplicate messages are ignored, as are some malformed ones), this has not been evaluated fully. Additionally, message authenticity (i.e. that a given message is actually coming from the sender in the "sender" field) is currently assumed to be handled outside of the protocol, by whatever networking code is shuttling messages around.
+
+
+## How to Build and Run
+
+You should try to build and run using the stable version of Rust instead of nightly. You can switch using the following:
+
+`rustup default stable` 
+`rustup update`
```
Comment (marsella) [inline README.md:57]: nit: The first command here does what you said (switch to stable); the second updates the version. I think it's still useful to have both, but consider updating the text above to be clear about that, e.g.
> The library requires a recent, stable version of Rust. You can switch to stable releases and update to the latest version using the following:

**Review by marsella — CHANGES_REQUESTED** [review]
Thanks for this PR! These are super useful additions. Most of my comments are sharing context that you didn't have, with some recommendations on making phrasing more specific.

File: README.md [hunk 5]
```diff
@@ -13,6 +13,11 @@ This codebase is generally intended to be network-agnostic. Programs take messag
 ## Project Dependencies
 This project relies on the `libpaillier` Rust crate using the GMP backend. GMP should be available during build-time.  
 
+### Rust Dependencies and Versions
+
+The preferred Rust version is cargo 1.68.2. 
+The preferred GMP version is 6.2.1.
+
```
[the remark this answers is not in the document]
Comment (marsella) [reply README.md:22]: @hridambasu Did you figure anything out about this?

[the remark this answers is not in the document]
Comment (hridambasu) [reply README.md:22]: I also think the latter that gmp does not auto install via the gmp-mpfr-sys crate. So shall I include this link for installation - https://formulae.brew.sh/formula/gmp?

[the remark this answers is not in the document]
Comment (marsella) [reply README.md:22]: Brew is only for OSX, so we need more than just that link. Unfortunately the gmp docs only have [instructions for downloading from source](https://gmplib.org/manual/Installing-GMP). [This site](http://www.mathemagix.org/www/mmdoc/doc/html/external/gmp.en.html) has some other Ubuntu commands for downloading the package and a separate option for OSX. Maybe the best approach is a little paragraph that's like, you can build from source \<link to GMP docs\> or using a package manager on linux \<link to mathemagix\> or OSX \<link to brew\>. The GMP website has [additional notes for other systems](https://gmplib.org/manual/Notes-for-Particular-Systems).

**Pull request merged**